crates/evm/core/src/backend/mod.rs

```diff
@@ -277,26 +275,6 @@ pub trait DatabaseExt: Database<Error = DatabaseError> {
     /// Marks the given account as persistent.
     fn add_persistent_account(&mut self, account: Address) -> bool;
 
-    /// Removes persistent status from all given accounts
-    fn remove_persistent_accounts(&mut self, accounts: impl IntoIterator<Item = Address>)
-    where
-        Self: Sized,
-    {
-        for acc in accounts {
-            self.remove_persistent_account(&acc);
-        }
-    }
-
-    /// Extends the persistent accounts with the accounts the iterator yields.
-    fn extend_persistent_accounts(&mut self, accounts: impl IntoIterator<Item = Address>)
```

Comment:
these can stay with "where Self: Sized"

Reply:
we need to use `dyn DatabaseExt` as `DB: DatabaseExt` generic and it doesn't work with `Self: Sized` methods in trait

Reply:
Ah it's that stupid thing with auto_impl, you have to add `#[auto_impl(keep_default_for(&, &mut, Rc, Arc, Box))]` to the function as well